test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -499,6 +499,7 @@ public void removeHeaderWarningAppender() {
 
     @Before
     public final void before() {
+        LeakTracker.setContextHint(getTestName());
```

Comment:
Doesn't the thread name already contain the test name, set by randomized runner?

Reply:
Yes but we're not guaranteed that all allocations happen from a test thread. In all the ITs we probably mostly care about allocations on `transport_worker` threads where the thread name isn't enough in isolation.

Reply:
Having a public method accessible anywhere in Elasticsearch, including plugins, makes me worried this will be used incorrectly. But I don't have an alternative at the moment...

Reply:
I don't think it's avoidable without adding lots of complexity. The tracker must live in prod code and we need to pass it the test name :) Only thing we could do is maybe move the tracker to the same package name that the `EsTestCase` lives under and make it package private?

Reply:
That won't work since server is modular, we can't have split packages.